plugins/chat/app/controllers/chat/emojis_controller.rb

```diff
@@ -3,7 +3,7 @@
 module Chat
   class EmojisController < ::Chat::BaseController
     def index
-      emojis = Emoji.all.group_by(&:group)
+      emojis = Emoji.allowed.group_by(&:group)
```

Comment:
It seems like `Emoji.allowed` was added just for this controller. Do you know if we can just continue to ship down `Emoji.all` but filter for the denied emojis on the client side when an emoji is used? This way, we get to avoid the [double caching](https://github.com/discourse/discourse/pull/20929/files#r1164764639) problem.

Reply:
> Alternatively we could go back to something similar to how I had this originally: use .all here but filter based on denied emojis (we can filter based on Emoji.deny this time though).

Yea maybe let's just do this for now and eat the 3ms overhead.